decidim-core/config/initializers/date_coercions.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+#
+# Overwrite Virtus default behaviour when coercing a String into a Date so that
+# we have the ability to use localized date formats.
+#
+module Coercible
+  class Coercer
+    # Coerce String values
+    class String < Object
+      def to_date(value)
+        # NOTE: The used format should be the same than the one used by
+        # the user (manual date entry) and the datepicker widget in the UI.
+        return value if value.blank?
+        ::Date.strptime(value, I18n.t("date.formats.decidim_short"))
+      rescue ArgumentError => e
+        raise ArgumentError, "#{e.message}: #{value}. Expected format :#{I18n.locale} #{I18n.t("date.formats.decidim_short")}"
```

Comment:
The situation where this error will be triggered this error is when a non-js user sends a date with an invalid format. In that case, our server will crash anyways and will get a 500 error anyways. Your patch could help when debugging the error, but errors like that just shouldn't happen. We should properly tell the user that the format is invalid instead.

Regarding how to solve it, it doesn't look easy and the fact that it's not easy in my opinion is raising a flag about the monkeypatch so deep down in the stack being added here.

I think it would be best if we went with your initial proposal: change the date(time) fields in forms to string fields and convert them to proper dates when passing them to the command through a reader method in the form.

Reply:
This was an improvement of the error message I was having in some tests, I find it interesting to keep it.

I agree that this monkeypatch is not desirable but I'll keep it until Rectify decides how they will proceed with the discontinuation of Virtus and solnic/coercible.

Reply:
So what's your proposal for the situation I'm describing? The error message will not only happen in some tests, it can be easily reproduced in real life! Leaving 500 errors opened for anyone to trigger whenever they want seems like a bad thing...

I figured out the proper way to use custom coercions in virtus without monkeypatching, and we're actually already using it in decidim. It's documented [here](https://github.com/solnic/virtus#custom-coercions), and we're using it [here](https://github.com/decidim/decidim/blob/5ac7b0878542b2bf760cde6be39406ba49e30944/decidim-core/lib/decidim/attributes/time_with_zone.rb). But, although an improvement, it would still suffer from the same issue as the current solution.

So, as I said before, the best way to do this is to change the columns to `String` type, and do the coercion at read time, when we can easily rescue potential errors and add them to the form. If we wanted to DRY up that dance, we could extract a `localizable_attribute` helper, in a similar manner as https://github.com/decidim/decidim/blob/5ac7b0878542b2bf760cde6be39406ba49e30944/decidim-core/lib/decidim/translatable_attributes.rb, but I don't think we need to do that in this PR.

Reply:
Actually, @tramuntanal, I just noticed that the current coercion in master raises the same argument error. I assumed `String#to_datetime` and family didn't raise on invalid dates and returned `nil` instead, but I was wrong. So this same error can be reproduced against the current master. So just forget about this comment, sorry :pray:.

Can you remove the monkeypatch and do this through a custom attribute, though?

Sorry for being a PITA, I wish I would've known all this stuff before, but I'm learning as we go.